### ISSUE TITLE
`select!` when returning an error gives an empty `expected`

### ISSUE THREAD

**Issue by ManicMarrc** [zesterer/chumsky#504]
Whenever the `select!` parser returns an `ExpectedFound` error, it makes the `expected` field empty.
I think this has to do with tokens that contains values like:

```rust
enum Token {
    Integer(u64),
    ...
}
```

**Comment by zesterer** [comment]
Yep, this is because chumsky has no way of understanding what `select!` is *trying* to expect. Right now, an empty `expected` field should be interpreted as 'something else'. If you want to give it a particular value, I recommend using `.labelled(...)`, which allows you to annotate parser patterns with explicit labels and have them appear in the `expected` list (for example, 'expected expression').

In the future we're planning to add more ways to automatically produce better error information.
